Optimize lock-bad-argument.html

Honry opened the pull request:
- centralize invalid lock type
 - simplified with promise_test
 - remove incorrect invaild_lock_type

<!-- Reviewable:start -->

<!-- Reviewable:end -->

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/23581)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/335445351)

Honry committed:
Optimize lock-bad-argument.html

 - centralize invalid lock type
 - simplified with promise_test
 - remove incorrect invaild_lock_type

Honry commented:
@marcoscaceres, PTAL.

Commit:
Prettier code, cleanup

promise_test(t => {
  const invalid_lock_types = [
    "invalid-orientation",
    null,
    undefined,
    123,
    window,
    ["portrait-primary", "landscape-primary"],
  ];
  const promisesToReject = invalid_lock_types.map(type =>
    promise_rejects(t, new TypeError(), screen.orientation.lock(type))
  );
  return Promise.all(promisesToReject);
}, "screen.orientation.lock() must throw given invalid input.");

promise_test(t => {
  return promise_rejects(t, new TypeError(), screen.orientation.lock());
}, "screen.orientation.lock() must throw when the input is missing.");

Commit:
prettier code, cleanup

marcoscaceres reviewed (approved):
Sorry about that first commit, I pasted my code into the wrong place lol!

marcoscaceres commented:
Your rewrite was way nicer than the original, btw... I just sprinkled a little bit of ✨.

Honry commented:
@marcoscaceres, you're nice. Thank you!
